target_postgres/target.py

```diff
@@ -96,8 +125,8 @@ def __init__(
             description=(
                 "SQLAlchemy connection string. "
                 + "This will override using host, user, password, port, "
-                + "dialect. Note that you must esacpe password special "
-                + "characters properly see "
+                + "dialect, and all ssl settings. Note that you must escape password "
```

Comment:
I never encountered a problem with it. That notice was present before I started on this PR.

It looks like we already escape special characters for users if provided for the `password` config option, just not when they use an SQLAlchemy URL. I don't think we should attempt to modify an SQLAlchemy URL since it creates ambiguity.

When a user passes an SQLAlchemyURL of `dialect+driver://username:alpha@bravo:1234/charlie@delta:4321/echo`, should that be resolved as
```
password=alpha
host=bravo
port=1234
database=charlie%40delta%3A4321%2Fecho
```
or
```
password=alpha%40bravo%3A1234%2Fcharlie
host=delta
port=4321
database=echo
```

The [SQLAlchemy documentation](https://docs.sqlalchemy.org/en/20/core/engines.html#escaping-special-characters-such-as-signs-in-passwords) asks for special characters to be escaped, and I think we should trust users to follow that specification.

Reply:
Let's leave it, I thought you added those words, it looks like it was as lint issue so I"ll just pass on thinking about it, I thought you hit something which is why there was a change here